go/vt/discovery/tablet_picker.go

```diff
@@ -336,73 +338,66 @@ func (tp *TabletPicker) PickForStreaming(ctx context.Context) (*topodatapb.Table
 				return nil, vterrors.Errorf(vtrpcpb.Code_CANCELED, "context has expired")
 			case <-timer.C:
 			}
+			// Got here? Means we iterated all tablets and did not find a
+			// viable candidate.
+			tp.incNoTabletFoundStat()
```

Comment:
Adding this line here seems wrong. We are incrementing the stat twice in the same if block, the first occurrence is on line 331.
It and the 2 lines of comments before it should be deleted. It made sense earlier to do it in two places because you could either
(a) find no candidates OR
(b) none of the candidates is reachable
With the change in this PR, the list of candidates only includes tablets that are already reachable and healthy

Reply:
Fixed. Thanks! ❤️